apps/core/client/management/get-checkout-url.ts

```diff
@@ -10,13 +16,65 @@ const RedirectUrlsSchema = z.object({
   }),
 });
 
+/*
+ * Use the Customer Login API to ensure the customer is logged into the hosted checkout during the checkout redirect
+ * process. This is a temporary solution to support logged-in customer checkout until the GraphQL API supports this redirect functionality.
+ *
+ * https://developer.bigcommerce.com/docs/store-operations/customers#customer-login-api
+ *
+ * @param checkoutUrl - The checkout URL to redirect to, from V3 Cart API
+ * @param customerId - The customer ID to log in as, from the session
+ * @returns The Customer Login API URL, with the redirect URL encoded in the JWT
+ */
+const wrapInCustomerLoginJwt = async (checkoutUrl: string, customerId: number) => {
+  if (!clientSecret || !clientId) {
+    // eslint-disable-next-line no-console
+    console.error(
+      'Missing BIGCOMMERCE_CLIENT_SECRET and/or BIGCOMMERCE_CLIENT_ID, unable to redirect to checkout while preserving login state',
+    );
+
+    return checkoutUrl;
+  }
+
+  const urlObj = new URL(checkoutUrl);
+  const origin = urlObj.origin;
+  const relativeUrl = urlObj.pathname + urlObj.search;
+
+  const secret = new TextEncoder().encode(clientSecret);
+  const alg = 'HS256';
+
+  const jwt = await new jose.SignJWT({
+    operation: 'customer_login',
+    store_hash: storeHash,
+    customer_id: customerId,
+    channel_id: channelId,
+    jti: crypto.randomUUID(),
+    redirect_to: relativeUrl,
+  })
+    .setProtectedHeader({ alg, typ: 'JWT' })
+    .setIssuer(clientId)
+    // accommodate minor clock skew, ensures the token is not created
+    // in the future from the BigCommerce server's perspective
+    .setIssuedAt(Date.now() / 1000 - 5)
+    .setExpirationTime('24h')
+    .sign(secret);
+
+  return `${origin}/login/token/${jwt}`;
+};
+
 // Url used to redirect the user to the checkout page
-export const getCheckoutUrl = async (cartId: string) => {
+export const getCheckoutUrl = async (cartId: string, customerId?: number) => {
```

Comment:
I think we should leave this method unchanged and move this customerLoginJwt logic closer to the server action.

Reply:
Yeah, I think baking the redirect URL into the page directly and then doing the JWT wrap at the last second (since these JWTs are very short-lived) would be a nice balance.

Reply:
That's what I did. You guys can feel free to reference this: https://github.com/CNanninga/bc-catalyst-contentful/blob/main/app/%5Blocale%5D/(default)/cart/_actions/redirect-to-checkout.ts